src/controllers/auth.controller.ts

```diff
@@ -161,4 +162,63 @@ export const githubAuthenticator = async (
   } catch (err: any) {
     passToExpressErrorHandler(err, next);
   }
-};
\ No newline at end of file
+};
+
+export const googleAuthenticator = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { googleAuthCode } = req.body;
+
+    const { name, email, googleId } = decodeGoogleCredential(googleAuthCode);
+    let user;
+
+    const alreadyExistingUser = await prisma.user.findFirst({
+      where: {
+        OR: [{ email }, { authType: "GOOGLE", authProviderId: googleId }],
+      },
+    });
+
+    if (
```

Comment:
this IF block can be simplified to
```
if (
  alreadyExistingUser &&
  email &&
  (
    !alreadyExistingUser.email ||
    alreadyExistingUser.authType !== "GOOGLE" ||
    alreadyExistingUser.authProviderId !== googleId ||
    alreadyExistingUser.name !== name
  )
) {
  ...
}
```
unless there is a need to check the existence of both authType and authProviderId simultaneously. and existence of email should also be separately checked since we're updating with it nonetheless.

Reply:
This if will be modified in the next feature depending on the coming logic so it's not a big deal right now